### ISSUE TITLE
RestoreRedundancyMergeSegments action is unable to make progress after OutrankedException

### ISSUE THREAD

**Issue by PavelZaytsev** [CorfuDB/CorfuDB#2521]
## Overview

If the seal of the servers occurred but the new layout is not yet committed during the RestoreRedundancyMergeSegments action, there is a chance of task not making any progress during the retries. This showcased in the following scenario: 

- There are nodes A, B and C. Nodes B, C belong to the cluster and node A does not.
- We are adding node A. Node A, B and C are at epoch 3. 
- Node A performs a state transfer and it finishes ok.
- Node A now wants to seal every server in the system (move every server in the system to epoch 4). 
- It needs only a quorum of seal responses to continue. Let's say it asynchronously seals servers A, B and C. B and C seal immediately and send the responses to A. A receives 2 out of 3 responses and continues. 
- Node A is now at epoch 3, Node B is at epoch 4 and Node C is at epoch 4.
- Node A forms a new layout (at epoch 4) where it is a part of a cluster, and now tries to attempt consensus on this layout. - Node A sends prepare request with a rank 1 to servers A, B and C. It needs only a quorum of prepare responses to continue the Paxos. 
- Node B at epoch 4 (recall that it was previously sealed) receives a prepare request at epoch 4. Sends a response -> ok.
- Node C at epoch 4 (recall that it was previously sealed) receives a prepare request at epoch 4. Sends a response -> ok.
- Node A at epoch 3 (it is not sealed yet, that's where the race is happening) is at epoch 3. Rejects a prepare request with a wrong epoch exception.  
- Node A continues with the second phase of Paxos - propose requests to all the nodes A, B and C since now it got the majority of prepare responses.
- Node B at epoch 4 (recall that it was previously sealed) receives a propose request at epoch 4. Sends a response -> ok.
- Node C at epoch 4 (recall that it was previously sealed) receives a propose request at epoch 4. Sends a response -> ok.
- Node A receives a propose request at epoch 4 (now it's sealed). It does not have a phase 1 rank but it got a propose request, so it throws an OuntrankedException. 
- Paxos fails.
- The cluster is in this state: 
  - Node A, B and C all have the layouts at epoch 3.
  - Node A, B and C are all sealed with epoch 4 and their routers are at epoch 4 too.
  - Primary sequencer is not bootstrapped. 
- Node A retries the entire workflow. One of the first steps is to retrieve a trim mark. Node A keeps on failing with a wrong epoch exception because it's runtime and the layout of every node is at epoch 3, but every server router is at epoch 4. 

TLDR: The issue arises because
1) The node that drives the RestoreRedundancyMergeSegments action is not guaranteed to be sealed before attemptConsensus.
2) A datapath logic and the layout reconfiguration logic are retried together.
